vulnerabilities/xxe/source/medium.php

```diff
@@ -0,0 +1,37 @@
+<?php
+function HandleXmlError($errno, $errstr, $errfile, $errline)
+        {
+                    if ($errno==E_WARNING && (substr_count($errstr,"DOMDocument::loadXML()")>0))
+                    {
+                        throw new DOMException($errstr);
+                    }
+                    else
+                        return false;
+        }
+
+function XXE(){
+	$xmloutput = '';
+	if( isset( $_FILES[ 'uploaded' ] ) ) {
+		$filepath = $_FILES[ 'uploaded' ][ 'tmp_name' ];
+		$content = file_get_contents($filepath);
+		libxml_disable_entity_loader(false);
```

Comment:
Even with this set, I'm still getting this error saying the box isn't vulnerable.

```
[Thu Dec 12 04:27:20.525636 2019] [proxy_fcgi:error] [pid 30085] [client 192.168.0.7:39702] AH01071: Got error 'PHP message: PHP Fatal error:  Uncaught DOMException: DOMDocument::loadXML(): Detected an entity reference loop in http://digininja.test/example.dtd, line: 2 in /var/sites/dvwa/non-secure/htdocs/vulnerabilities/xxe/source/medium.php:6\nStack trace:\n#0 [internal function]: HandleXmlError(2, 'DOMDocument::lo...', '/var/sites/dvwa...', 22, Array)\n#1 /var/sites/dvwa/non-secure/htdocs/vulnerabilities/xxe/source/medium.php(22): DOMDocument->loadXML('<?xml version="...', 6)\n#2 /var/sites/dvwa/non-secure/htdocs/vulnerabilities/xxe/index.php(49): XXE()\n#3 {main}\n  thrown in /var/sites/dvwa/non-secure/htdocs/vulnerabilities/xxe/source/medium.php on line 6', referer: http://dvwa.test/vulnerabilities/xxe/
```